src/OpenTelemetry.Instrumentation.AWSLambda/OpenTelemetry.Instrumentation.AWSLambda.csproj

```diff
@@ -8,12 +8,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Amazon.Lambda.APIGatewayEvents" Version="2.4.1" />
-    <PackageReference Include="Amazon.Lambda.Core" Version="1.2.0" />
-    <PackageReference Include="Amazon.Lambda.SNSEvents" Version="2.0.0" />
-    <PackageReference Include="Amazon.Lambda.SQSEvents" Version="2.1.0" />
+    <PackageReference Include="Amazon.Lambda.APIGatewayEvents" Version="2.7.0" />
+    <PackageReference Include="Amazon.Lambda.Core" Version="2.2.0" />
```

Comment:
Is raising the lower bound of these dependencies necessary? In particular this one from major version 1 to 2?

Reply:
These versions all correlate to the versions that first introduced support for native AoT: [CHANGELOG](https://github.com/aws/aws-lambda-dotnet/blob/master/RELEASE.CHANGELOG.md#release-2023-10-26).

Version 2.1.0 of Amazon.Lambda.Core is the first version that added explicit support for `net6.0`, which was required by the changes asked for in #1545, and the only release since then was the 2.2.0 release for `net8.0` support that resolves the native AoT warnings.

Based on that, I think these are reasonable increases to require of users while also making this assembly usable without trim warnings in native AoT applications. There were also already breaking changes listed in the CHANGELOG for this library before this PR or #1545.

Thoughts @normj?

Reply:
> while also making this assembly usable without trim warnings in native AoT applications

I wonder, because I assume the following:

1. Users of the instrumentation _should_ still have their own dependencies on Lambda packages and not rely on transitive dependencies here (unfortunately no implementation/api distinction exists in Nuget like in Java Gradle modules)
2. If users want AoT, they can & should upgrade their own dependencies as needed. The library we release only specifies a lower bound.

Based on what you write with the .NET 6 support, an upgrade to 2.1 is totally reasonable without question, but I wonder about 2.2.

Reply:
I can downgrade it a bit in my native AoT Lambda using .NET 8 and see if I get any IL warnings (https://github.com/martincostello/alexa-london-travel/pull/1020) or not.

My library does already specify its own dependencies for the Lambda assemblies [it needs](https://github.com/martincostello/alexa-london-travel/blob/d82a420645cfc52e0c9f3615bf3a7665e52ae5b8/Directory.Packages.props#L6-L8), _but_ I don't use API Gateway, SNS or SQS, but I get all of those dependencies via this library transiently. Needing the end-user to manually upgrade the transient dependencies to get it to work properly for native AoT isn't ideal.

Reply:
>  but I don't use API Gateway, SNS or SQS, but I get all of those dependencies via this library transiently

Good point. This is not ideal anyway. I don't think there is a way to have these as compile-only dependencies and then check at runtime if they are actually present?

Reply:
Norm might know some tricks for dealing with that as the core AWS SDKs have to do similar things for things like STS authentication via the built-in fallback credential providers, but yeah that's a bit of a spiral out of the original scope here 😅.

Reply:
Sorry I can't think of any special tricks here. The only difference between 2.1 and 2.2 for Amazon.Lambda.Core is updates to the project file to have the .NET 8 target and mark the package trimmable for .NET 8 if that makes you feel better taking the updated dependency.